deku-p/src/core/stdlib/parallel.mli

```diff
@@ -7,6 +7,7 @@ module Worker : sig
   type t = worker
 
   val make : domains:#Eio.Domain_manager.t -> sw:Eio.Switch.t -> worker
+  val teardown : worker -> unit
```

Comment:
I understand why this was done, but it is quite bad as now you may have a copy of a worker that is not working, so some piece of code may try to schedule work and never receive the result.

But you can achieve the same by killing the ~sw in make, this will make everything on that switch dies and you can control granularity by making smaller switches.

Reply:
I see. I'll keep generating more tests for now like we discussed offline, but I'll make sure to address this in a future PR.